src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Unix.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class RuntimeInformation
+    {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                s_osDebugName = Interop.Sys.GetUnixVersion();
+            }
+
+            return s_osDebugName;
+        }
+
+        public static Architecture GetOSArchitecture()
+        {
+            if (null == s_osArch)
+            {
+                Interop.Sys.ProcessorArchitecture arch = (Interop.Sys.ProcessorArchitecture)Interop.Sys.GetUnixArchitecture();
+                switch (arch)
+                {
+                    case Interop.Sys.ProcessorArchitecture.ARM:
+                        s_osArch = Architecture.ARM;
+                        break;
+
+                    case Interop.Sys.ProcessorArchitecture.x64:
+                        s_osArch = Architecture.x64;
+                        break;
+
+                    case Interop.Sys.ProcessorArchitecture.x86:
+                        s_osArch = Architecture.x86;
```

Comment:
Let me think out loud here,

So enum has 2 operations
```c#
{
    value = set_value;
    hasValue = true;
}
```

So there might be a case, where one thread has set value, but not updated the hasValue, which means the other thread will read value as null, when it is not null, but that is ok, since setting the value again, will be the same value each time. But I don't understand how the reverse might happen, when hasValue = true, and value = null, this can happen if the instructions in the above enum is not in order. Is there a way where the compiler will parallelize independent operations, where ```hasValue = true``` will be executed before ```value = set_value```?